docs: Electron 32 blog (#629)
electron/website: blog/electron-32-0.md

---
title: Electron 32.0.0
date: 2024-08-20T00:00:00.000Z
authors: VerteDinde
slug: electron-32-0
tags: [release]
---

Electron 32.0.0 has been released! It includes upgrades to Chromium `128.0.6613.36`, V8 `12.8`, and Node `20.16.0`.

---

The Electron team is excited to announce the release of Electron 32.0.0! You can install it with npm via `npm install electron@latest` or download it from our [releases website](https://releases.electronjs.org/releases/stable). Continue reading for details about this release.

If you have any feedback, please share it with us on [Twitter](https://twitter.com/electronjs) or [Mastodon](https://social.lfx.dev/@electronjs), or join our community [Discord](https://discord.com/invite/electronjs)! Bugs and feature requests can be reported in Electron's [issue tracker](https://github.com/electron/electron/issues).

## Notable Changes

### Highlights

> - Added new documentation for API version history, which was created by @piotrpdev as part of Google Summer of Code. [#42982](https://github.com/electron/electron/pull/42982)

Author: Note for reviewers: we have a draft blog post by @piotrpdev describing the new API history feature: https://github.com/electron/website/pull/627/files Depending on merge order, we should link that post here as well 😄

> - Added new API version history in our documentation, a feature created by @piotrpdev as part of Google Summer of Code. [#42982](https://github.com/electron/electron/pull/42982)

Author: Note for reviewers: we have a draft blog post by @piotrpdev describing the new API history feature: https://github.com/electron/website/pull/627/files Depending on merge order, we should link that post here as well 😄

Author: Quick note: We're going to amend this post after publishing this - this post will publish first, then Piotr's blog 👍

- Removed nonstandard File.path extension from the Web File API. [#42053](https://github.com/electron/electron/pull/42053)
- Aligned failure pathway in File System Access API with upstream when attempting to open a file or directory in a blocked path. [#42993](https://github.com/electron/electron/pull/42993)

> - Added the following existing navigation related APIs to `webcontents.navigationHistory`: `canGoBack`, `goBack`, `canGoForward`, `goForward`, `canGoToOffset`, `goToOffset`, `clear`. The previous navigation APIs are not deprecated. [#41752](https://github.com/electron/electron/pull/41752)

Reviewer: ```suggestion
- Added the following existing navigation-related APIs to `webcontents.navigationHistory`: `canGoBack`, `goBack`, `canGoForward`, `goForward`, `canGoToOffset`, `goToOffset`, `clear`. The previous navigation APIs are not deprecated. [#41752](https://github.com/electron/electron/pull/41752)
```

### Stack Changes

- Chromium`128.0.6613.36`
  - [New in 128](https://developer.chrome.com/blog/new-in-chrome-128/)
  - [New in 127](https://developer.chrome.com/blog/new-in-chrome-127/)
- Node `20.16.0`
  - [Node 20.16.0 blog post](https://nodejs.org/en/blog/release/v20.16.0/)
- V8 `12.8`

Electron 32 upgrades Chromium from `126.0.6478.36` to `128.0.6613.36`, Node from `20.14.0` to `20.16.0`, and V8 from `12.6` to `12.8`.

### New Features

> - Added support for responding to auth requests initiated from utility process. [#43317](https://github.com/electron/electron/pull/43317)

Reviewer: ```suggestion
- Added support for responding to auth requests initiated from the utility process via the `app` module's [`'login'`](https://www.electronjs.org/docs/latest/api/app) event. [#43317](https://github.com/electron/electron/pull/43317)
```

add details on how to do this without needing to read the original PR

> - Added `cumulativeCPUUsage` to `AppMetrics` and `CPUUsage`. [#41819](https://github.com/electron/electron/pull/41819)

Reviewer: ```suggestion
- Added the `cumulativeCPUUsage` property to the `CPUUsage` structure, which returns the total seconds of CPU time used since process startup. [#41819](https://github.com/electron/electron/pull/41819)
```

`AppMetrics` isn't the name of a structure in Electron, so I think we should remove references to it.

`app.getAppMetrics` returns an array of `ProcessMetrics` and then that contains a `CPUUsage` prop.

- Added the following existing navigation related APIs to `webcontents.navigationHistory`: `canGoBack`, `goBack`, `canGoForward`, `goForward`, `canGoToOffset`, `goToOffset`, `clear`. [#41752](https://github.com/electron/electron/pull/41752)
- Extended `WebContentsView` to accept pre-existing `webContents` object. [#42086](https://github.com/electron/electron/pull/42086)
- Added a new property `prefersReducedTransparency` to `nativeTheme`, which indicates whether the user has chosen to reduce OS-level transparency via system accessibility settings. [#43137](https://github.com/electron/electron/pull/43137)
- Aligned failure pathway in File System Access API with upstream when attempting to open a file or directory in a blocked path. [#42993](https://github.com/electron/electron/pull/42993)

> - Enabled the Windows Control Overlay API on Linux. [#42681](https://github.com/electron/electron/pull/42681)

Author: Done here!: https://github.com/electron/electron/pull/43388

> - Enabled `zstd` compression in net http requests. [#43300](https://github.com/electron/electron/pull/43300)

Author: I think it's all network requests, will amend
